x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.tsx

```diff
@@ -126,6 +130,36 @@ export const EndpointList = () => {
   const dispatch = useDispatch<(a: EndpointAction) => void>();
   // cap ability to page at 10k records. (max_result_window)
   const maxPageCount = totalItemCount > MAX_PAGINATED_ITEM ? MAX_PAGINATED_ITEM : totalItemCount;
+  const [showTransformFailedCallout, setShowTransformFailedCallout] = useState(false);
+
+  const getTransformStats = useCallback(async () => {
+    if (!services?.http) {
+      return;
+    }
+
+    const transformStats: TransformStatsResponse = await services.http.get(TRANSFORM_STATS_URL);
+    if (!transformStats?.transforms) {
+      return;
+    }
+
+    const hasFailure = transformStats.transforms.some((transform) =>
+      WARNING_TRANSFORM_STATES.has(transform?.state)
+    );
+
+    setShowTransformFailedCallout(hasFailure);
+  }, [services.http]);
+
+  useEffect(() => {
+    if (!endpointsExist || !listData?.length) {
+      return;
+    }
+
+    getTransformStats();
```

Comment:
If this is going to be executed each time endpointsExists and listData is updated, could this be moved to the middleware? This will keep the component clean with less business logic. You can use the redux store to keep the state (stats response)